### DISCUSSION TITLE
zh/guide/drivers/SMB

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#120] [Announcements]
# zh/guide/drivers/SMB

是把smb挂载到Alist，而不是Alist通过smb共享出来，而不是smb能像webdav共享出来。 因无相关设备和服务，无法测试，可以根据 GitHub Discussions 讨论中提供的办法自行测试一下 https://github.com/alist-org/alist/discussions/2215#discussioncomment-4...

https://alist.nn.ci/zh/guide/drivers/SMB.html

**Comment by fayfoxcat** [comment]
有打算支持smb共享么？

**Reply by jiabingxian** [reply]
***同感***

**Comment by Hanxu4131** [comment]
SMB 需要使用445、139端口。我目前使用Linux Ubuntu搭建的nas，两个硬盘做软Raid1。我的运营商关闭了我的21FTP、139SMB端口，但为我保留了445端口。我目前可以在公网通过SMB连接到我的服务器，但在Alist的SMB中配置时，提示我504错误。我为我家中的IP做了域名访问，目前为止我可以理解到，Alist的SMB服务正常做了域名解析，因为他已经报告了我准确的IP地址，但无论我手动设置445端口，或是使用SMB默认，都无法连接。多数为以下错误信息“Failed create storage in database: UNIQUE constraint failed: x_storages.mount_path”

**Reply by Hanxu4131** [reply]
还有以下内容“dial tcp 117.9.179.63:445: connect: connection timed out”

**Reply by Hanxu4131** [reply]
我的IP地址是动态地址，这个地址会失效，不要用这个进行连接测试了。我的Linux一直在弹出不同IP地址的连接失败。我想更应该解决目前的SMB服务。我家中的服务器是Raid1（SMB+Alist）使用。Ubuntu使用的SMB3，而家中还有Airport时间返回舱，所使用的为SMB1服务。这两个我都进行了测试，Alist目前都无法连接，我没有SMB2的设备，如果有SMB2的测试结果请联系我

**Comment by Azroys** [comment]
请问会支持CIFS协议吗

**Comment by zzerosama** [comment]
好想alist能直接支持smb和ftp啊，smb是因为mxplayer只支持smb，安卓13对于文件管理器挂载阿里云webdav再播放不行，必须保持挂载的软件在前台，要不几秒钟就卡住进度，文件管理器小窗就可以，但是没有播放记录又阻挡视频；想用ftp是因为喜欢filezilla的保留传输文件的时间戳功能，webdav却不支持这个功能😂

**Comment by fashamu** [comment]
没有写入权限，怎么回事呀

**Comment by qzrsa** [comment]
请问这是什么问题
failed get objs: failed to list objs: dial tcp ************00:445: connect: connection refused

**Comment by fashamu** [comment]
应该是smb共享文件夹丢失了，检查了smb主机，重启好了
________________________________
发件人: 丶曲K人散g ***@***.***>
发送时间: 2023年10月4日 12:57
收件人: alist-org/docs ***@***.***>
抄送: fashamu ***@***.***>; Comment ***@***.***>
主题: Re: [alist-org/docs] zh/guide/drivers/SMB (Discussion #120)


请问这是什么问题
failed get objs: failed to list objs: dial tcp ************00:445: connect: connection refused

―

**Comment by Perter1999** [comment]
有大佬帮忙看看吗？我这个提交了还报错，装docker里面的，电脑都能直接访问。
报错：Failed init storage: connection error: invalid transport format

**Comment by heierhu** [comment]
教程地址:
http://ikwiki.top:31860
玩客云拓展→alist_smb挂载

**Comment by dth188** [comment]
我现在的环境是软路由架设alist服务器，局域网内有nas设备，软路由上还拖了一块2t的硬盘，为了达成一站式服，把各种网盘挂载到alist服务器的同时，把软路由的2t硬盘作为本地存储挂载到了Alist服务器，并且把nas以smb协义的方式也挂载到了Alist服务器。外网通过动态域名端口转发可以直接登录内网alist服务器，形成了各种存储一站式服务。由于宽带上传速度只有50兆，在外网访问内网过程中，除了那些通过302代理的网盘以外网实际网速和网盘提供下载网速决定外，其他通过本地代理的存储方式都能跑满50兆。现在要向网友们通报的是在内网环境中使用alist服务器。本人的内网是软路由作为主路由，2.5g设备和1G设备混用。各种网盘使用正常，软路由自带2t存储能跑满速，但是nas的smb存储降速严重。通报的原因可能是套娃结构中协议不兼容产生的降速，比如nas的数据要到软路由中转一下 ，但本身这台nas就连在软路由上，正常情况下也是需要通过软路由中转，但是可以满速。怀疑是在软路由的alist服务器又做了一次本地代理。相当于这个数据在这台软路由上进行了两次转发？

**Comment by Perter1999** [comment]
大佬们帮忙看看。我保存的时候报错，电脑都是可以正常访问的，环境是安装到docker里面的，版本v3.36.0，docker也安装了smbclient(./sbin/apk  add samba-client)
报错：Failed init storage: connection error: invalid transport format

**Reply by zzerosama** [reply]
这是直接把电脑的配置备份，还原到docker上的吧，我以前也老是出现这个，把所有的用户名删掉重新创建就好了，不知道你这是不是这样

**Comment by wheat0r** [comment]
alist代理smb的性能很差，两个smb共享，一个跑到60M/s，一个差不多45M/s。
直连这两个smb都能跑到100M/s以上。
